apisix/plugins/slslog/rfc5424.lua

```diff
@@ -0,0 +1,107 @@
+local LOG_INFO      = 6       --  informational
+local LOG_DEBUG     = 7       --  debug-level messages
+
+local LOG_KERN      = 0       --  kernel messages
+local LOG_USER      = 1       --  random user-level messages
+local LOG_MAIL      = 2       --  mail system
+local LOG_DAEMON    = 3       --  system daemons
+local LOG_AUTH      = 4       --  security/authorization messages
+local LOG_SYSLOG    = 5       --  messages generated internally by syslogd
+local LOG_LPR       = 6       --  line printer subsystem
+local LOG_NEWS      = 7       --  network news subsystem
+local LOG_UUCP      = 8       --  UUCP subsystem
+local LOG_CRON      = 9       --  clock daemon
+local LOG_AUTHPRIV  = 10      --  security/authorization messages (private)
+local LOG_FTP       = 11      --  FTP daemon
+local LOG_LOCAL0    = 16      --  reserved for local use
+local LOG_LOCAL1    = 17      --  reserved for local use
+local LOG_LOCAL2    = 18      --  reserved for local use
+local LOG_LOCAL3    = 19      --  reserved for local use
+local LOG_LOCAL4    = 20      --  reserved for local use
+local LOG_LOCAL5    = 21      --  reserved for local use
+local LOG_LOCAL6    = 22      --  reserved for local use
+local LOG_LOCAL7    = 23      --  reserved for local use
+
+local Facility = {
+    ["KERN"] = LOG_KERN,
+    ["USER"] = LOG_USER,
+    ["MAIL"] = LOG_MAIL,
+    ["DAEMON"] = LOG_DAEMON,
+    ["AUTH"] = LOG_AUTH,
+    ["SYSLOG"] = LOG_SYSLOG,
+    ["LPR"] = LOG_LPR,
+    ["NEWS"] = LOG_NEWS,
+    ["UUCP"] = LOG_UUCP,
+    ["CRON"] = LOG_CRON,
+    ["AUTHPRIV"] = LOG_AUTHPRIV,
+    ["FTP"] = LOG_FTP,
+    ["LOCAL0"] = LOG_LOCAL0,
+    ["LOCAL1"] = LOG_LOCAL1,
+    ["LOCAL2"] = LOG_LOCAL2,
+    ["LOCAL3"] = LOG_LOCAL3,
+    ["LOCAL4"] = LOG_LOCAL4,
+    ["LOCAL5"] = LOG_LOCAL5,
+    ["LOCAL6"] = LOG_LOCAL6,
+    ["LOCAL7"] = LOG_LOCAL7,
+}
+
+local Severity = {
+    ["EMEGR"] = LOG_EMERG,
+    ["ALERT"] = LOG_ALERT,
+    ["CRIT"] = LOG_CRIT,
+    ["ERR"] = LOG_ERR,
+    ["WARNING"] = LOG_WARNING,
+    ["NOTICE"] = LOG_NOTICE,
+    ["INFO"] = LOG_INFO,
+    ["DEBUG"] = LOG_DEBUG,
+}
+local os_date = os.date
+local ngx = ngx
+local string_format = string.format
+local rfc5424_timestamp_format = "!%Y-%m-%dT%H:%M:%S.000Z"
+local rfc5424_format = "<%d>1 %s %s %s %d - [logservice project=\"%s\" logstore=\"%s\"" ..
+                    " access-key-id=\"%s\" access-key-secret=\"%s\"] %s\n"
+
+local _M = { version = 0.1 }
+
+function _M.encode(facility, severity, hostname, appname, pid, project,
+    logstore, access_key_id, access_key_secret, msg)
+    local pri = (Facility[facility] * 8 + Severity[severity])
+    ngx.update_time()
+    local t = os_date(rfc5424_timestamp_format, ngx.now())
+
+    if not hostname then
+        hostname = "-"
+    end
+
+    if not appname then
+        appname = "-"
+    end
+
+    return string_format(rfc5424_format, pri, t, hostname, appname, pid, project,
+     logstore, access_key_id, access_key_secret, msg)
```

Comment:
bad indentation

Reply:
fix

Reply:
I do not think is my format is right, I reference this doc: https://github.com/apache/apisix/blob/master/CODE_STYLE.md#fromHistory